jsparagus/gen.py

```diff
@@ -1571,53 +1582,1724 @@ def load(cls, filename):
+        if isinstance(term, Action):
+            for t, s in self.epsilon:
+                if t == term:
+                    return s
+            raise KeyError(term)
+        elif isinstance(term, Nt):
+            return self.nonterminals[term]
+        if isinstance(term, ErrorSymbol):
+            return self.errors[term]
+        else:
+            return self.terminals[term]
+
+    def get(self, term, default):
+        try:
+            return self.__getitem__(term)
+        except KeyError:
+            return default
+
+    def __str__(self):
+        conflict = ""
+        if self.is_inconsistent():
+            conflict = " (inconsistent)"
+        return "{}{}:\n{}".format(self.index, conflict, "\n".join([
+            "\t{} --> {}".format(k, s) for k, s in self.edges()]))
+
+    def __eq__(self, other):
+        assert isinstance(other, StateAndTransitions)
+        if sorted(self.locations) != sorted(other.locations):
+            return False
+        if sorted(self.delayed_actions) != sorted(other.delayed_actions):
+            return False
+        return True
+
+    def __hash__(self):
+        return self._hash
+
+
+def on_stack(grammar, term):
+    """Returns whether an element of a production is consuming stack space or
+    not."""
+    if isinstance(term, Nt):
+        return True
+    elif grammar.is_terminal(term):
+        return True
+    elif isinstance(term, LookaheadRule):
+        return False
+    elif isinstance(term, ErrorSymbol):
+        return True
+    elif isinstance(term, End):
+        return True
+    elif term is NoLineTerminatorHere:
+        # No line terminator is a property of the next token being shifted. It
+        # is implemented as an action which once shifted past the next term,
+        # will check whether the previous term shifted is on a new line.
+        return False
+    raise ValueError(term)
+
+def callmethods_to_funcalls(expr, pop, ret, depth, funcalls):
+    # TODO: find a way to carry alias sets to here.
+    alias_set = ["parser"]
+    if isinstance(expr, int):
+        stack_index = pop - expr
+        if depth == 0:
+            expr = FunCall("id", alias_set, alias_set, (stack_index,), ret)
+            funcalls.append(expr)
+            return ret
+        else:
+            return stack_index
+    elif isinstance(expr, Some):
+        res = callmethods_to_funcalls(expr.inner, pop, ret, depth, funcalls)
+        return Some(res)
+    elif expr is None:
+        return None
+    elif isinstance(expr, CallMethod):
+        def convert_args(args):
+            for i, arg in enumerate(args):
+                yield callmethods_to_funcalls(arg, pop, ret + "_{}".format(i), depth + 1, funcalls)
+        args = tuple(convert_args(expr.args))
+        expr = FunCall(expr.method, alias_set, alias_set, args, ret)
+        funcalls.append(expr)
+        return ret
+    elif expr == "accept":
+        expr = FunCall("accept", alias_set, alias_set, (), ret)
```

Comment:
Accept is special

Reply:
I will open a follow-up issue after merging.

Reply:
https://github.com/mozilla-spidermonkey/jsparagus/issues/356 is created to fix this issue.